### PR TITLE
[url_launcher] Fixes for Dart plugin registry

### PR DESCRIPTION
**Pull request by blasten**
Required for https://github.com/flutter/flutter/issues/52267

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/url_launcher/url_launcher/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 6.0.0-nullsafety.6
+
+* Fixes for Dart plugin registry.
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/CHANGELOG.md:3]: While we should fix the typos, url_launcher's desktop implementations don't need a Dart registry, so this description doesn't really make sense.

File: packages/url_launcher/url_launcher_linux/lib/url_launcher_linux.dart [hunk 2]
```diff
@@ -1,3 +1,11 @@
 // The url_launcher_platform_interface defaults to MethodChannelUrlLauncher
 // as its instance, which is all the Linux implementation needs. This file
 // is here to silence warnings when publishing to pub.
+
+/// UrlLauncherLinux provides the registerWith() method.
+class UrlLauncherLinux {
+  /// Binds the implementation to the interface.
+  static void registerWith() {
+    // todo
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_linux/lib/url_launcher_linux.dart:9]: There's nothing to do here; I don't see any purpose this method could ever serve (per the comment above).

File: packages/url_launcher/url_launcher_linux/pubspec.yaml [hunk 3]
```diff
@@ -1,12 +1,14 @@
 name: url_launcher_linux
 description: Linux implementation of the url_launcher plugin.
-version: 0.1.0-nullsafety.3
+version: 0.1.0-nullsafety.4
 homepage: https://github.com/flutter/plugins/tree/master/packages/url_launcher/url_launcher_linux
 
 flutter:
   plugin:
+    implements: url_launcher
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_linux/pubspec.yaml:8]: Did this ever end up being used anywhere? I remember that it was in the spec, so we should do it, but is something reading it?

File: packages/url_launcher/url_launcher_linux/pubspec.yaml [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
     platforms:
       linux:
+        dartPluginClass: UrlLauncherLinux
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_linux/pubspec.yaml:11]: Remove, per comment in the other file.

File: packages/url_launcher/url_launcher_macos/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
+## 0.1.0-nullsafety.3
+
+* Fixes for Dart plugin registry.
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_macos/CHANGELOG.md:3]: Same comments throughout.

File: packages/url_launcher/url_launcher_windows/CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,7 @@
+## 0.1.0-nullsafety.3
+
+* Fixes for Dart plugin registry.
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_windows/CHANGELOG.md:3]: And here.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Many of the changes here don't really make sense; path_provider_{windows,linux} and shared_preferences_{windows,linux} are the plugins that need the registry.

File: packages/url_launcher/url_launcher_windows/pubspec.yaml [hunk 7]
```diff
@@ -3,13 +3,15 @@ description: Windows implementation of the url_launcher plugin.
...
+version: 0.1.0-nullsafety.3
 homepage: https://github.com/flutter/plugins/tree/master/packages/url_launcher/url_launcher_windows
 
 flutter:
   plugin:
+    implements: url_launcher
     platforms:
       windows:
+        dartPluginClass: UrlLauncherWindows
         pluginClass: UrlLauncherPlugin
```
Comment (Sunbreak) [inline packages/url_launcher/url_launcher_windows/pubspec.yaml:15]: If `dartPluginClass` takes response, maybe `pluginClass` could be removed?

**Comment by blasten** [comment]
Closing as this isn't relevant anymore. Will follow up with a smaller fix for the typo in the pubspec.

**Pull request closed without merging**